src/usr/share/netcheck/netcheck.py

```diff
@@ -776,3 +794,55 @@ def _calculate_next_log_time(self, loop_time):
         Returns the datetime that the logging should occur.
         """
         return loop_time + datetime.timedelta(seconds=self.config['periodic_status_delay'])
+
+    def _check_for_gateway_change(self):
+        """Checks the current state of the default gateway and issues a broadcast if it is
+        different from the prior state.
+        """
+
+        new_gateway_state = self._get_default_gateway_state()
+
+        if new_gateway_state != self.default_gateway_state:
+            self.logger.info(
+                'The default gateway has changed to %s via %s on interface %s.',
+                new_gateway_state['address'],
+                new_gateway_state['connection_id'],
+                new_gateway_state['interface'])
+
+            self.default_gateway_state = new_gateway_state
+            self.broadcaster.issue()
+
+    def _get_default_gateway_state(self):
+        """Retrieves information about the current default gateway.
+
+        Returns a dictionary containing a gateway IP address, interface name, and associated
+        connection id.
+        """
+        gateway_state = None
+
+        with pyroute2.IPRoute() as ip_route:
+            default_routes = ip_route.get_default_routes()
+            if default_routes:
+                gateway_state = {
+                    'address': None,
+                    'interface': None,
+                    'connection_id': None}
```

Comment:
Putting this all on one line was fine, but you can leave it like this if you want. Either is fine.

Reply:
This format is much easier for me to read, so I'd like to keep it this way.